zfilterfactory.h

```diff
@@ -0,0 +1,96 @@
+/*
+*  zfilterfactory.h
+*  PHD Guiding
+*
+*  Created by Ken Self
+*  Copyright (c) 2018 Ken Self
+*  All rights reserved.
+*
+*  This source code is distributed under the following "BSD" license
+*  Redistribution and use in source and binary forms, with or without
+*  modification, are permitted provided that the following conditions are met:
+*    Redistributions of source code must retain the above copyright notice,
+*     this list of conditions and the following disclaimer.
+*    Redistributions in binary form must reproduce the above copyright notice,
+*     this list of conditions and the following disclaimer in the
+*     documentation and/or other materials provided with the distribution.
+*    Neither the name of openphdguiding.org nor the names of its
+*     contributors may be used to endorse or promote products derived from
+*     this software without specific prior written permission.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+*  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+*  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+*  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+*  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+*  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+*  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+*  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+*  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+*  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*
+*/
+
+#ifndef ZFILTER_FACTORY_H_INCLUDED
+#define ZFILTER_FACTORY_H_INCLUDED
+#include <complex>
+
+enum FILTER_DESIGN
+{
+    BESSEL,
+    BUTTERWORTH,
+    CHEBYCHEV,    
+};
+
+class ZFilterFactory
+{
+public:
+    std::vector<double> xcoeffs, ycoeffs;
+    double gain() { return ::hypot(dc_gain.imag(), dc_gain.real()); };
+    double corner() { return 1 / raw_alpha1; };
+    int order() { return m_order;  };
+    ZFilterFactory( FILTER_DESIGN f, int o, double p );
+private:
+    const double  PI = 3.14159265358979323846;  // Microsoft C++ does not define M_PI ! 
```

Comment:
M_PI is defined in <math.h> in MS C++
can remove this definition and include <math.h>